_blog/2018/v1alpha3-routing.md

```diff
@@ -0,0 +1,347 @@
+now configure the (virtual) destination itself, with all of its rules in an ordered list within a corresponding
+`VirtualService` resource. For example, where previously we had two `RouteRule` resources for the
+[Bookinfo]({{home}}/docs/guides/bookinfo.html) application’s `reviews` service, like this:
+
+```yaml
+apiVersion: config.istio.io/v1alpha2
+kind: RouteRule
+metadata:
+  name: reviews-default
+spec:
+  destination:
+    name: reviews
+  precedence: 1
+  route:
+  - labels:
+      version: v1
+---
+apiVersion: config.istio.io/v1alpha2
+kind: RouteRule
+metadata:
+  name: reviews-test-v2
+spec:
+  destination:
+    name: reviews
+  precedence: 2
+  match:
+    request:
+      headers:
+        cookie:
+          regex: "^(.*?;)?(user=jason)(;.*)?$"
+  route:
+  - labels:
+      version: v2
+```
+
+In `v1alph3`, we provide the same configuration in a single `VirtualService` resource:
+
+```yaml
+apiVersion: networking.istio.io/v1alpha3
+kind: VirtualService
+metadata:
+  name: reviews
+spec:
+  hosts:
+    - reviews
+  http:
+  - match:
+    - headers:
+        cookie:
+          regex: "^(.*?;)?(user=jason)(;.*)?$"
+    route:
+    - destination:
+        name: reviews
+        subset: v2
+  - route:
+    - destination:
+        name: reviews
+        subset: v1
+```
+
+As you can see, both of the rules for the `reviews` service are consolidated in one place, which at first may or may not
+seem preferable. However, if you look closer at this new model, you’ll see there are fundamental differences that make
+`v1alpha3` vastly more functional.
+
+First of all, notice that the destination service for the `VirtualService` is specified using a `hosts` field (repeated field,
+in fact) and is then again specified in a `destination` field of each of the route specifications. This is a very important
+difference from the previous model.
+
+A `VirtualService` describes the mapping between one or more user-addressable destinations to the actual destination workloads inside the mesh. In our example, they are the same, however, the user-addressed hosts can be any DNS
+names with optional wildcard prefix or CIDR prefix that will be used to address the service. This can be particularly
+useful in facilitating turning monoliths into a composite service built out of distinct microservices without requiring the
+consumers of the service to adapt to the transition.
+
+For example, the following rule allows users to address both the reviews and ratings services of the Bookinfo application
+as if they are parts of a bigger (virtual) service at `http://bookinfo.com/`:
+
+```yaml
+apiVersion: networking.istio.io/v1alpha3
+kind: VirtualService
+metadata:
+  name: bookinfo
+spec:
+  hosts:
+    - bookinfo.com
+  http:
+  - match:
+    - uri:
+        prefix: /reviews
+    route:
+    - destination:
+        name: reviews
+  - match:
+    - uri:
+        prefix: /ratings
+    route:
+    - destination:
+        name: ratings
+  ...
```

Comment:
I suggest removing k8s-style-documentation dots. Usually they signify that there is more to yaml-formatted command output than shown in example e.g. case `kubectl describe`. This will remedy a plethora of copy-paste issues.